### PR TITLE
Fix usage of the tool. Provide -h and --help

### PR DESCRIPTION
**Pull request by posix4e**

### CODE REVIEW

File: src/bin/rustfmt.rs [hunk 1]
```diff
@@ -27,29 +27,47 @@ fn main() {
...
     def_config_file.read_to_string(&mut def_config).unwrap();
     let config = Box::new(Config::from_toml(&def_config));
+
     let (args, write_mode) = determine_params(std::env::args());
 
     run(args, write_mode, config);
 
     std::process::exit(0);
 }
+fn usage<S: Into<String>>(reason: S) {
```
Comment (cassiersg) [inline src/bin/rustfmt.rs]: nit: missing blank line

File: src/bin/rustfmt.rs [hunk 2]
```diff
@@ -27,29 +27,47 @@ fn main() {
...
+    panic!("{}\n\r usage: rustfmt [-h Help] [--write-mode=[true/false]] <file_name>", reason.into())
```
Comment (cassiersg) [inline src/bin/rustfmt.rs]: This should rather `println!` and then `std::process::exit`.

**Comment by cassiersg** [comment]
Thanks for this PR. Rustfmt is becoming a well-manered tool !
Just a few comments in-line.
There is also a rebase needed.

File: src/bin/rustfmt.rs (hunk 2 above)
Comment (posix4e) [inline src/bin/rustfmt.rs]: Will fix

**Comment by posix4e** [comment]
Ok rebased it and fixed the nits

**Comment by cassiersg** [comment]
Nice !

**Pull request merged**